### DISCUSSION TITLE
Does MicroStream support Kotlin?

### DISCUSSION THREAD

**Discussion by vngantk** [microstream-one/microstream#274] [Q&A] [answered]
We are migrating our backend software from Java to Kotlin, and are considering using MicroStream as our data storage mechanism. We assume that MicroStream should work well with Kotlin because of its compatibility with Java. Any limitations and issues we need to be aware of?

**Comment by fh-ms** [comment, marked as answer]
The only currently known limitation is that the immutable collections of Kotlin are not supported. We need to provide special TypeHandlers for these. But, besides that, Kotlin works fine.

**Reply by rocketraman** [reply]
> the immutable collections of Kotlin are not supported

What immutable collections? Do you mean the *read-only* collections like `List`? Generally these are just backed by a standard `ArrayList` on the JVM.

**Comment by vngantk** [comment]
Thank you for your prompt response.

Regards,
VN
